modules/renter/hostdb/hosttree/hosttree.go

```diff
@@ -0,0 +1,290 @@
+package hosttree
+
+import (
+	"crypto/rand"
+	"errors"
+	"sync"
+
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+var (
+	// ErrWeightTooHeavy is returned from a Fetch() call if a weight that exceeds
+	// the total weight of the tree is requested.
+	ErrWeightTooHeavy = errors.New("requested a too-heavy weight")
+
+	// ErrNegativeWeight is returned from an Insert() call if an entry with a
+	// negative weight is added to the tree. Entries must always have a positive
+	// weight.
+	ErrNegativeWeight = errors.New("cannot insert using a negative weight")
+
+	// ErrNilEntry is returned if a fetch call results in a nil tree entry. nodes
+	// should always have a non-nil entry, unless they have been Delete()ed.
+	ErrNilEntry = errors.New("node has a nil entry")
+
+	// ErrHostExists is returned if an Insert is called with a public key that
+	// already exists in the tree.
+	ErrHostExists = errors.New("host already exists in the tree")
+
+	// ErrNoSuchHost is returned if Remove is called with a public key that does
+	// not exist in the tree.
+	ErrNoSuchHost = errors.New("no host with specified public key")
+)
+
+type (
+	// WeightFunc is a function used to weight a given HostDBEntry in the tree.
+	WeightFunc func(modules.HostDBEntry) types.Currency
+
+	// HostTree is a data structure that contains a weighted tree of
+	// `HostEntries`, and provides methods for inserting, retreiving, and
+	// modifing entries.
+	HostTree struct {
+		root *node
+
+		// hosts is a map of public keys to nodes.
+		hosts map[string]*node
+
+		// wf applies a weight to a hostEntry
+		wf WeightFunc
+
+		mu sync.Mutex
```

Comment:
`wf` is not an acceptable name, because it's not a common acronym. `mu` and `tg` are special cases because they are ubiquitous throughout the project - pretty much every module has both a `mu` and a `tg`. But only the host tree has a weightFunc, so it should be spelled out.

Also, probably okay to group the weightFunc and mu together.